connectors/acoustic/acoustic/acoustic_client.py

```diff
@@ -0,0 +1,299 @@
+        """
+
+        auth_endpoint = "oauth/token"
+        url = f"{self.base_url}/{auth_endpoint}"
+
+        _grant_type = "refresh_token"
+        _request_body = {
+            "grant_type": _grant_type,
+            "client_id": self._client_id,
+            "client_secret": self._client_secret,
+            "refresh_token": self._refresh_token,
+        }
+
+        headers = {
+            "content-type": "application/x-www-form-urlencoded",
+        }
+
+        response = requests.post(url, headers=headers, data=_request_body)
+        response.raise_for_status()
+
+        return response.json()["access_token"]
+
+    def get_agg_metric_org(
+        self, request_template: str, template_params: dict
+    ) -> List[dict]:
+        """
+        Extracts a listing of Acoustic Campaign emails that are sent for an organization
+        for a specified date range and provides metrics for those emails.
+
+
+        :param request_template: path to XML template file containing request body template to be used
+        :param template_params: values that should be used to render the request template provided
+
+        :return: Returns back a list of rows of data returned by the API call as a dictionary
+
+        More information about the specific API call:
+        https://developer.goacoustic.com/acoustic-campaign/reference/reporting#getaggregatetrackingfororg
+        """
+
+        with open(request_template, "r") as _file:
+            request_body_template = jinja2.Template(_file.read())
+
+        request_body = request_body_template.render(**template_params)
+
+        request = {
+            "url": f"{self.base_url}/{self.XML_API_ENDPOINT}",
+            "headers": {
+                "content-type": "text/xml;charset=utf-8",
+                "authorization": f"Bearer {self._access_token}",
+            },
+            "data": request_body,
+        }
+
+        response = requests.post(**request)
+        response.raise_for_status()
+
+        data = xmltodict.parse(response.text)
+        result_data = data["Envelope"]["Body"]["RESULT"]
+
+        if not result_data["SUCCESS"].lower() == "true":
+            """
+            Example response when there is no mailing info found
+            OrderedDict(
+                [('Envelope', OrderedDict([('Body', OrderedDict(
+                    [('RESULT', OrderedDict(
+                        [('SUCCESS', 'false')])), ('Fault', OrderedDict(
+                            [('Request', None), ('FaultCode', None), ('FaultString', 'No mailings.'), ('detail', OrderedDict(
+                                [('error', OrderedDict([('errorid', '156'), ('module', None), ('class', 'SP.API'), ('method', None)]))])
+                            )])
+                        )])
+                    )])
+                )]
+            )
+            """  # noqa: E501
+            print(result_data["Fault"])
+            if result_data["Fault"]["FaultCode"].lower():
```

Comment:
Might be reading this wrong but what's the `False/None` value here? I think there'll be an `AttributeError` if there's no `FaultCode` key